packages/odyssey-react/src/components/Infobox/Infobox.module.scss

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) 2018-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+.infobox {
+  @include border-radius($base-border-radius);
+
+  display: grid;
+  grid-template-areas:
+    'icon title'
+    '. content'
+    '. actions';
+  grid-template-columns: max-content 1fr;
+  width: 100%;
+  max-width: calc(#{$max-line-length} + #{$spacing-m} + #{$spacing-m});
+  padding-block: $spacing-m;
+  padding-inline: $spacing-m;
+  column-gap: $spacing-s;
+
+  a {
```

Comment:
Can you talk me through the design use case here with links? Does this need to be an indirect descendant or can we scope this down a bit more to use only direct descendants, e.g. `> a`

Reply:
You are right to be concerned! The use case here is overriding link styles in instances like this: 

![CleanShot 2021-08-06 at 14 30 04@2x](https://user-images.githubusercontent.com/62259644/128572995-d0edc6c1-67c9-47e2-8dd3-e6f99a3ccb4d.png)

If I were to shift to doing the Link CSS modules prior to merging this I could change this to something like the following: 

```
a {
  composes: link from '../Link/Link.module.scss';
  
  color: $text-body;
  text-decoration: underline;
}
```

@jeffbelser-okta @edburyenegren-okta Do you guys think this is a workable path forward?

Reply:
We can drill down on the exact implementation, but I like this as a starting point.

To @jeffbelser-okta's point above, our original overrides anticipate the Link styles being applied to all `a`s, but that's not a safe assumption. It also anticipates the use of `a` (not Link) within an Infobox as user/author-supplied content. In an ideal world, we'd provide proper display in both cases, especially since we currently support mixed-content for messages on the design side:

<img width="442" alt="Screen Shot 2021-08-06 at 3 14 16 PM" src="https://user-images.githubusercontent.com/36284167/128576090-2a30adf1-957f-49e7-9780-f448dd085e8a.png">

Maybe in the future, we could restrict the action to an Infobox.Link slot rather than allowing authors to include an `a` in the messaging content?

Even in cases where we can limit support to Link and not `a`, we'll want to find an agreeable path. See [this issue](https://github.com/css-modules/css-modules/issues/147) for a years-long exploration of different approaches to proper contextual styling with CSS Modules.

Reply:
If we need to style links in this way we have two options that will prevent us from implicit coupling or contextual CSS:

we can provide a new variant for the `<Link />` component and provide guidance to use that specific variant when placing it as a child of `Infobox` something like `variant="infobox"` or similar.

or alternatively we can expose high order grouping of components, `InfoboxWithlink` which takes care of the specific styles for all its children.

With where we are today the former option feels like the correct path forward. We need to move away from contextual styles during this refactor effort.